File: Backend/Nodejs/src/routes/artist.ts
import { Router } from "express";
import multer from "multer";
import { ArtistController } from "../controllers/artist";
import { s3Storage } from "../libs/s3client";

export const artistRouter = Router();
const uploadArtist = multer({ storage: s3Storage({ userType: "artist" }) });

artistRouter.get("/", ArtistController.getArtists);
artistRouter.get("/:id", ArtistController.getArtist);
artistRouter.get("/:id/songs", ArtistController.getSongs);
artistRouter.get("/:id/songs/notInAlbum", ArtistController.getSongsNotInAlbum);
artistRouter.post("/newArtist", uploadArtist.single("avatar"),ArtistController.createArtist);
artistRouter.patch("/:id", ArtistController.editArtist);
artistRouter.delete("/:id", ArtistController.deleteArtist);

File: Backend/Python/src/routes/artist.py
from flask import Blueprint
from src.controllers.artist import ArtistController

artist_blueprint = Blueprint("blueprint", __name__)

artist_blueprint.route("/", methods=["GET"])(ArtistController.get_artists)
artist_blueprint.route("/<int:id>", methods=["GET"])(ArtistController.get_artist)
artist_blueprint.route("/<int:id>/songs", methods=["GET"])(ArtistController.get_songs)
artist_blueprint.route("/<int:id>/songs/notInAlbum", methods=["GET"])(ArtistController.song_not_album)
artist_blueprint.route("/newArtist", methods=["POST"])(ArtistController.create_artist)
artist_blueprint.route("/", methods=["PATCH"])(ArtistController.edit_artist)
artist_blueprint.route("/<int:id>", methods=["DELETE"])(ArtistController.delete_artist)
